mlos_bench/mlos_bench/storage/base_storage.py

```diff
@@ -209,6 +230,9 @@ def _setup(self) -> None:
 
             This method is called by `Storage.Experiment.__enter__()`.
             """
+            self._status = Status.RUNNING
```

Comment:
Maybe add some asserts on expected status to check for invalid state transitions.
Wouldn't be terrible to document the expected state transitions in a README.md or docstring either.